matron/src/oracle.c

```diff
@@ -484,136 +496,141 @@ void o_restart_audio() {
-void o_tape_level(float level) {
-    lo_send(ext_addr, "/tape/level", "f", level);
+void o_set_level_tape(float level) {
+    lo_send(crone_addr, "/set/level/tape", "f", level);
 }
 
-void o_tape_new(char *file) {
-    lo_send(ext_addr, "/tape/newfile", "s", file);
+void o_tape_rec_open(char *file) {
+    lo_send(crone_addr, "/tape/record/open", "s", file);
 }
 
-void o_tape_start_rec() {
-    lo_send(ext_addr, "/tape/start_rec", NULL);
+void o_tape_rec_start() {
+    lo_send(crone_addr, "/tape/record/start", NULL);
 }
 
-void o_tape_pause_rec() {
-    lo_send(ext_addr, "/tape/pause_rec", NULL);
+void o_tape_rec_stop() {
+    lo_send(crone_addr, "/tape/record/stop", NULL);
 }
 
-void o_tape_stop_rec() {
-    lo_send(ext_addr, "/tape/stop_rec", NULL);
+void o_tape_play_open(char *file) {
+    lo_send(crone_addr, "/tape/play/open", "s", file);
 }
 
-void o_tape_open(char *file) {
-    lo_send(ext_addr, "/tape/openfile", "s", file);
+void o_tape_play_start() {
+    lo_send(crone_addr, "/tape/play/start", NULL);
 }
 
-void o_tape_play() {
-    lo_send(ext_addr, "/tape/play", NULL);
+void o_tape_play_stop() {
+    lo_send(crone_addr, "/tape/play/stop", NULL);
 }
 
-void o_tape_pause() {
-    lo_send(ext_addr, "/tape/pause", NULL);
+
+//--- cut
+void o_cut_enable(int i, float value) {
+    lo_send(crone_addr, "/set/enabled/cut", "if", i, value);
 }
 
-void o_tape_stop() {
-    lo_send(ext_addr, "/tape/stop", NULL);
+void o_set_level_adc_cut(float value) {
+    lo_send(crone_addr, "/set/level/adc_cut", "f", value);
 }
 
-//--- aux effects controls
-// enable / disable aux fx processing
-void o_set_aux_fx_on() {
-    lo_send(crone_addr, "/set/enabled/reverb", "f", 1);
+void o_set_level_ext_cut(float value) {
+    lo_send(crone_addr, "/set/level/ext_cut", "f", value);
 }
 
-void o_set_aux_fx_off() {
-    lo_send(crone_addr, "/set/enabled/reverb", "f", 0);
+void o_set_level_cut_aux(float value) {
+    lo_send(crone_addr, "/set/level/cut_aux", "f", value);
 }
 
+void o_set_level_cut(int index, float value) {
+	lo_send(crone_addr, "/set/level/cut", "if", index, value);
+}
 
-//--- insert effects controls
-void o_set_insert_fx_on() {
-    lo_send(crone_addr, "/set/enabled/compressor", "f", 1);
+void o_set_level_cut_cut(int src, int dest, float value) {
+	lo_send(crone_addr, "/set/level/cut_cut", "iif", src, dest, value);
 }
 
-void o_set_insert_fx_off() {
-    lo_send(crone_addr, "/set/enabled/compressor", "f", 0);
+void o_set_pan_cut(int index, float value) {
+	lo_send(crone_addr, "/set/pan/cut", "if", index, value);
 }
 
-void o_set_insert_fx_mix(float value) {
-    lo_send(crone_addr, "/set/level/ins_mix", "f", value);
+void o_set_cut_param(const char* name, int voice, float value) {
+    static char buf[128];
+    sprintf(buf, "/set/param/cut/%s", name);    
+    lo_send(crone_addr, buf, "if", voice, value);
 }
 
+void o_set_level_input_cut(int src, int dst, float level) {
+    lo_send(crone_addr, "/set/level/in_cut", "iif", src, dst, level);
+}
 
```

Comment:
i haven't really verified that it actually works to "overload" an OSC method like this; it should probably just be a separate command.

Reply:
this separates them:

https://github.com/monome/norns/pull/671/commits/80c48ee372bd0f0f3bd9740f1c2382a8349d9bfc